### PR TITLE
Update CPM entries during post-release version bump PR

### LINKED ISSUE #14146: Release pipeline version bump PR should also update Central Package Management (Directory.Packages.props)
## Problem

When the .NET release pipeline runs and a package is published, the automated post-release version bump PR (created by `archetype-sdk-publish-net.yml`) only updates two files:

1. **The `.csproj` file** — via `eng/scripts/Increment-dotnet-package.ps1`
2. **The `CHANGELOG.md`** — via `eng/common/scripts/Update-Changelog.ps1`

It does **not** update the Central Package Management (CPM) file (e.g. `Directory.Packages.props`), which means a second, manual follow-up PR is required to keep CPM in sync with the new version. This costs additional developer time and can lead to version drift if the manual update is delayed or forgotten.

## Desired Behavior

The automated version bump PR should also update the relevant CPM file (`Directory.Packages.props` or equivalent) as part of the same PR, so that a single automated PR covers all version references after a release.

## Affected Files

- Pipeline template: `eng/pipelines/templates/stages/archetype-sdk-publish-net.yml` (lines 105-132)
- Version increment script: `eng/scripts/Increment-dotnet-package.ps1`
- Changelog update script: `eng/common/scripts/Update-Changelog.ps1`

## Suggested Approach

Add a new step (or extend `Increment-dotnet-package.ps1`) to locate and update the `PackageVersion` entry in the CPM props file that corresponds to the package being released. This would run between the existing csproj/changelog updates and the `create-pull-request` step so all changes are included in the same commit and PR.

### PR DESCRIPTION
**Pull request by m-nash**
## Summary

Extends \Increment-dotnet-package.ps1\ to also update Central Package Management (CPM) files when the automated version bump PR is created after a release. This eliminates the need for a separate manual PR to update CPM, saving developer time and preventing version drift.

### \ng/scripts/Increment-dotnet-package.ps1\
After bumping the csproj version and updating the changelog, the script now:
1. Extracts the NuGet package name from the csproj (\PackageId\ > \AssemblyName\ > filename)
2. Scans for CPM files in both the new (\ng/centralpackagemanagement/*.Packages.props\) and old (\ng/Packages.Data.props\) locations
3. Updates all \PackageVersion\/\PackageReference\ entries matching the package to the released version
4. Skips override files (\overrides/\ subdirectory), MSBuild property references, and version ranges
5. Gracefully handles missing CPM files (no-op if none exist)

The CPM update is wrapped in its own try/catch — failures surface as pipeline warnings but never block the version bump PR.

### \ng/scripts/tests/Increment-dotnet-package.Tests.ps1\ (new)
34 Pester tests covering all edge cases including: prefix-safe matching (\Azure.Core\ vs \Azure.Core.Amqp\), regex-safe dots, property references, version ranges, multiple entries, mixed formats, empty files, whitespace variations, and more.

## Compatibility
Works before and after https://github.com/Azure/azure-sdk-for-net/pull/55609 (CPM modernization). If no CPM files exist, the script silently skips — zero behavior change for repos that haven't adopted CPM.

Fixes #14146

### CODE REVIEW

File: eng/scripts/Increment-dotnet-package.ps1 [hunk 1]
```diff
@@ -102,6 +102,93 @@ function Update-CsprojVersion {
...
+
+function Get-CentralPackageManagementFiles {
+    param (
+        [Parameter(Mandatory = $true)] [string]$RepoRoot
+    )
+    $files = @()
+
+    # New CPM structure (post CPM migration): eng/centralpackagemanagement/*.Packages.props
+    # Only files directly in this directory — NOT in overrides/ subdirectory
+    $cpmDir = Join-Path $RepoRoot "eng" "centralpackagemanagement"
```
Comment (weshaggard) [inline eng/scripts/Increment-dotnet-package.ps1:148]: What context to you plan to use this in? Does this path exist anywhere?

File: eng/scripts/Increment-dotnet-package.ps1 [hunk 2]
```diff
@@ -102,6 +102,93 @@ function Update-CsprojVersion {
...
+        [Parameter(Mandatory = $true)] [string]$NewVersion,
+        [Parameter(Mandatory = $true)] [object[]]$Files
+    )
+
+    $totalUpdates = 0
+    $escapedName = [regex]::Escape($PackageName)
+
+    # Match PackageVersion (Include|Update) and PackageReference (Update) entries with literal versions.
+    # The version must start with a digit to skip MSBuild property references like $(SomeVersion).
+    $pattern = '(<Package(?:Version|Reference)\s+(?:Include|Update)="' + $escapedName + '"\s+Version=")(\d[^"]*?)(")'
```
Comment (weshaggard) [inline eng/scripts/Increment-dotnet-package.ps1:175]: IIRC there are some packages that exist more than once with different conditions on their item group. Do you want update all if they exist?

File: eng/scripts/Increment-dotnet-package.ps1 [hunk 3]
```diff
@@ -118,6 +205,35 @@ try {
 
   Update-CsprojVersion -CsprojPath $csprojPath -OldVersion $currentVersion -NewVersion $newVersion
 
+  # Update Central Package Management files with the released version so
```
Comment (weshaggard) [inline eng/scripts/Increment-dotnet-package.ps1:208]: Where is this Increment-dotnet-package.ps1 script even called?

Comment (weshaggard) [reply eng/scripts/Increment-dotnet-package.ps1:208]: https://github.com/search?q=repo%3AAzure%2Fazure-sdk-for-net%20Increment-dotnet-package.ps1&type=code turned up nothing. It looks like it might be used in the tools repo only.

Comment (weshaggard) [reply eng/scripts/Increment-dotnet-package.ps1:208]: I'm guessing you might want https://github.com/Azure/azure-sdk-for-net/blob/main/eng/scripts/Update-PkgVersion.ps1 which is what is used in the .NET repo. 

Another question is do we always want to increment the version? We didn't do that originally because we wanted to have .NET continue to target the lowest version needed and not always be looking at the latest. However, if you and Jesse agree with this, I'm good with changing that.

File: eng/scripts/Increment-dotnet-package.ps1 (hunk 1 above)
Comment (m-nash) [reply eng/scripts/Increment-dotnet-package.ps1:148]: This path doesn't exist today — it will be created when https://github.com/Azure/azure-sdk-for-net/pull/55609 merges (the CPM modernization PR). The script checks `Test-Path` before scanning, so if the directory doesn't exist it's a no-op. This is intentional so the script works both before and after that PR lands.

The old path (`eng/Packages.Data.props`) on line 155 is what exists today.

File: eng/scripts/Increment-dotnet-package.ps1 (hunk 2 above)
Comment (m-nash) [reply eng/scripts/Increment-dotnet-package.ps1:175]: Yes — the regex `[regex]::Replace()` replaces **all** occurrences in each file, not just the first. So if a package appears in multiple `<ItemGroup>` blocks with different conditions, all of them get updated. We have a test specifically for this case: *'with multiple entries of same package in one file (different ItemGroups)'* which asserts both occurrences are updated.

This is safe after the changes https://github.com/Azure/azure-sdk-for-net/pull/55609

File: eng/scripts/Increment-dotnet-package.ps1 (hunk 3 above)
Comment (m-nash) [reply eng/scripts/Increment-dotnet-package.ps1:208]: It's called from `eng/pipelines/templates/stages/archetype-sdk-publish-net.yml` line 111 — the post-release 'Increment version in csproj' step that runs after a .NET tool package is published to NuGet.

**Comment by m-nash** [comment]
Closing — the version bump script that needs updating is actually in azure-sdk-for-net (eng/pipelines/templates/stages/archetype-net-release.yml), not this repo. Moving the fix there. Issue #14146 should remain open.
